src/app/containers/StoryPromo/IndexAlsos/index.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useContext } from 'react';
+import { oneOfType, arrayOf, shape, number, string } from 'prop-types';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+import pathOr from 'ramda/src/pathOr';
+import MediaIndicator from '@bbc/psammead-media-indicator';
+import {
+  IndexAlsos,
+  IndexAlso,
+  IndexAlsosUl,
+  IndexAlsosLi,
+} from '@bbc/psammead-story-promo/index-alsos';
+import { ServiceContext } from '../../../contexts/ServiceContext';
+
+const IndexAlsosContainer = ({ alsoItems, script, service }) => {
+  const {
+    translations: { media: mediaTranslations },
+  } = useContext(ServiceContext);
+
+  const getMediaType = (cpsType, mediaType) => {
+    const isPGL = cpsType === 'PGL';
+    const isMedia = cpsType === 'MAP';
+    const media = mediaType || 'Video';
+
+    if (!isPGL && !isMedia) {
+      return null;
+    }
+
+    const type = isPGL ? 'photogallery' : media.toLowerCase();
+
+    return type;
+  };
+
+  const IndexAlsosMediaIndicator = (cpsType, mediaType) => {
+    const indexAlsosMediaType = getMediaType(cpsType, mediaType);
+
+    return indexAlsosMediaType ? (
+      <MediaIndicator service={service} type={indexAlsosMediaType} indexAlsos />
+    ) : null;
+  };
+
+  return (
+    <IndexAlsos offScreenText="Related content">
+      {alsoItems.length > 1 ? (
+        <IndexAlsosUl>
+          {alsoItems.slice(0, 3).map(item => {
+            const { id, cpsType, mediaType } = item;
+            const headline = pathOr(null, ['headlines', 'headline'], item);
+            const url = pathOr(null, ['locators', 'assetUri'], item);
+
+            return (
+              <IndexAlsosLi
+                key={id}
+                script={script}
+                service={service}
+                url={url}
+                mediaIndicator={IndexAlsosMediaIndicator(
```

Comment:
Could we store this in a variable and pass it in. Instead of calling the function in both conditions.